app/frontend/src/features/connections/useCancelFriendRequest.ts

```diff
@@ -0,0 +1,32 @@
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Error } from "grpc-web";
+import { useQueryClient, useMutation } from "react-query";
+import { service } from "../../service";
+import { SetMutationError } from "./friends";
+
+interface CancelFriendRequestVariables {
+  friendRequestId: number;
+  setMutationError: SetMutationError;
+}
+
+export default function useCancelFriendRequest() {
+  const queryClient = useQueryClient();
+  const { mutate: cancelFriendRequest, isLoading, isSuccess } = useMutation<
+    Empty,
+    Error,
+    CancelFriendRequestVariables
+  >(({ friendRequestId }) => service.api.cancelFriendRequest(friendRequestId), {
+    onMutate: async ({ setMutationError }) => {
+      setMutationError("");
+      await queryClient.cancelQueries("friendRequestsSent");
+    },
+    onSuccess: () => {
```

Comment:
```suggestion
    onSuccess: () => {
      queryClient.removeQueries(["friendRequestsSent", friendRequestId], true);
```

Reply:
The API returns all friend requests in a single list, rather than individually and requiring `useQueries`, so don't think this makes sense